wire-compiler/src/main/java/com/squareup/wire/WireCompiler.kt

```diff
@@ -126,6 +130,11 @@ class WireCompiler internal constructor(
       targets += SwiftTarget(
         outDirectory = swiftOut
       )
+    } else if (customOut != null) {
+      targets += CustomTarget(
+        outDirectory = customOut,
+        schemaHandlerFactory = newSchemaHandler(schemaHandlerFactoryClass!!)
+      )
```

Comment:
Maybe we do the check here like this:

```suggestion
    } else if (customOut != null || schemaHandlerFactoryClass != null) {
      if (customOut == null || schemaHandlerFactoryClass == null) {
        throw IllegalArgumentException("Both custom_out and schema_handler_factory_class need to be set")
      }
      targets += CustomTarget(
        outDirectory = customOut,
        schemaHandlerFactory = newSchemaHandler(schemaHandlerFactoryClass)
      )
```